packages/common/src/util/http.js

```diff
@@ -0,0 +1,135 @@
+import { Client, MockAgent } from 'undici';
+
+const clients = new Map();
```

Comment:
I may have messed up this pattern :grimacing: 

This assumes that the adaptor is loaded into an isolate context, and that each workflow has its own context.

A fair assumption for CLI and for platform.

But moving forward we are going to be using long-running adaptors. Ignoring the short term details, really the adaptor should be cached in memory and re-used across different workflows. If we assume we can do this securely (TBD), we also need to be able to do it stably.

The common client map is a problem here. It should be initialised once per workflow, and destroyed when the workflow ends.

In an adaptor, we can hook to execute to re-initialise the clientmap. Although if two jobs are running concurrently they will interfere. Alternatively in an adaptor we can use the state object - a pattern I've been against, but would be totally safe. But the request helper doesn't see state.

The best solution I can see right now is for the util to have a `destroyClient(baseUrl, options)` function, which the adaptor MUST call  at the end of the job. That would ensure clients are not re-used (well maybe - actually, if two jobs are running concurrently with the same URL and options, they'll interfere)

Perhaps this is all nonsense and it's up to the runtime engine to ensure that each workflow has a dedicated module instance. Disabling any module re-use is surely the best way to ensure stability, flexibility AND security across workflows, and would allow the adaptor to do whatever the heck it wants (which is ideally the policy we would adopt, rather than being very prescriptive).

Ok, so this is a long comment to say that it is appropriate for the new runtime engine to support this pattern.